securitybot/bot.py

```diff
@@ -266,16 +304,25 @@ def handle_new_tasks(self):
         '''
         for task in self.tasker.get_new_tasks():
             # Log new task
+            task = self._store_or_update_active_task(task)
             logging.info('Handling new task for {0}'.format(task.username))
-
             self._add_task(task)
 
     def handle_in_progress_tasks(self):
         # type: () -> None
         '''
         Handles all in progress tasks.
         '''
-        pass
+        now = datetime.now()
+        for task in self.tasker.get_active_tasks():
+            task = self._store_or_update_active_task(task)
+            logging.info('Handling in-progress task {0}'.format(task))
+            delta = now - task.event_time
+            for escalation in task.escalation:
+                if escalation.escalated_at is None and delta.seconds > escalation.delay_in_sec:
```

Comment:
added `is_notified`, `set_notified` and `should_notify` functions, and removed the previous ones. Also renamed the property to indicate it's 'privateness' 👍